### DISCUSSION TITLE
Can we delete some keys dynamically to reduce the size of returning object of `getStaticProps`?

### DISCUSSION THREAD

**Discussion by txfs19260817** [i18next/next-i18next#2071] [Q&A]
Hi there, I have some large translation JSON files and next.js may complain [`large-page-data`](https://nextjs.org/docs/messages/large-page-data) error because `getStaticProps` returns too much data. So do we have a built-in solution to trim some unused keys to tackle this issue?

Here is my workaround that modifies `initialI18nStore` directly:

https://github.com/txfs19260817/falinks-teambuilder/blob/main/src/pages/pastes/%5Bid%5D.tsx#L37-L88

Thanks!

**Comment by adrai** [comment]
This is all what's built-in: https://github.com/i18next/next-i18next#5-declaring-locale-dependencies
